Lombiq.Tests.UI/Extensions/OrchardFeaturesTestingUITestContextExtensions.cs

```diff
@@ -0,0 +1,199 @@
+            contentItemsPage
+                .CreateNewPage()
+                    .Title.Set("Audit Trail Test Page")
+                    .Publish.ClickAndGo()
+                .AlertMessages.Should.Contain(message => message.IsSuccess);
+
+            await context.GoToAdminRelativeUrlAsync(auditTrailPath);
+
+            context.Exists(ByHelper.TextContains("of the Page"));
+
+            context.Exists(ByHelper.TextContains("was published"));
+
+            context.Exists(ByHelper.TextContains("was created"));
+        });
+
+    public static Task TestWorkflowsAsync(this UITestContext context) =>
+        context.ExecuteTestAsync(
+            "Test Workflows",
+            async () =>
+            {
+                var workflowsPath = "/Workflows/Types";
+                var contentItemPublishTestSuccessMessage = "The content item was published.";
+
+                await context.EnableFeatureDirectlyAsync("OrchardCore.Workflows");
+                await context.GoToAdminRelativeUrlAsync(workflowsPath + "/EditProperties");
+
+                await context.ClickAndFillInWithRetriesAsync(By.Id("Name"), "Test workflow");
+                await context.ClickReliablyOnSubmitAsync();
+
+                await context.ClickReliablyOnAsync(By.XPath("//button[@data-activity-type='Event']"));
+                await context.ClickReliablyOnAsync(By.XPath("//a[contains(@href, 'ContentPublishedEvent')]"));
+
+                await context.ClickAndFillInWithRetriesAsync(By.Id("IActivity_Title"), "Content Published Trigger");
+                await context.SetCheckboxValueAsync(By.XPath("//input[@value='Page']"), isChecked: true);
+                await context.ClickReliablyOnSubmitAsync();
+
+                await context.ClickReliablyOnAsync(By.XPath("//button[@data-activity-type='Task']"));
+                await context.ClickReliablyOnAsync(By.XPath("//a[contains(@href, 'NotifyTask')]"));
+
+                await context.ClickAndFillInWithRetriesAsync(By.Id("IActivity_Title"), "Content Published Notification");
+                await context.ClickAndFillInWithRetriesAsync(By.Id("NotifyTask_Message"), contentItemPublishTestSuccessMessage);
+                await context.ClickReliablyOnSubmitAsync();
+
+                var taskXPath = "//div[contains(@class, 'activity-task')]";
+
+                context.DragAndDropToOffset(By.XPath(taskXPath), 400, 0);
+
+                context.DragAndDrop(
+                    By.XPath("//div[@class = 'jtk-endpoint jtk-endpoint-anchor jtk-draggable jtk-droppable']"), // #spell-check-ignore-line
+                    By.XPath(taskXPath));
+
+                await context.ClickReliablyOnAsync(By.XPath("//div[contains(@class, 'activity-event')]"));
+
+                // Waiting is necessary here because when we click on the event, we need to wait for the popup
+                // otherwise, the startup task button won't be clicked.
+                await Task.Delay(1000);
```

Comment:
Never do this. Wait for the popup's element to appear explicitly with `Exists()`.

Reply:
The elment already exists it's just not visible, if we add the visibility check too, it still won't work, since it exists, but I guess it's not enough time, to be properly loaded in. 
But anyway, I came up with a simple solution. Saving the workflow once, then clicking on the button works, then saving again.

Reply:
Exists() repeats until a timeout.

Reply:
I understand, but it passes right away yet the button still cannot be clicked, that's why I went with the solution I wrote.